### ISSUE TITLE
Training iteration stops silently if not Debugging

### ISSUE THREAD

**Issue by iustin94** [tensorflow/tensorflow#18802]
Have I written custom code: Yes
OS Platform and Distribution: Windows server 2012 R2 Standard
TensorFlow installed from: Pip repositories
TensorFlow version: 1.7.0
Bazel version: NA
CUDA/cuDNN version: NA
GPU model and memory: NA
Exact command to reproduce NA

I'm attempting to train a tensorflow.contrib.keras.model.Sequencial on an image dataset. I have noticed however that when running the script normally, everything appears to work fine until the process does not exit. When I ran the scrip in debug mode, I received the message "StopIteration: Could not import PIL.Image. The use of 'array_to_img' requires PIL". The error is pretty self-explanatory, however is it intended behavior to fail silently like this?  I did not realize at first that the execution halted without exiting, until sometime passed.

This is my model initialization and training block:
 
        model = tf.contrib.keras.models.Sequential()
        model.add(tf.contrib.keras.layers.Conv2D(32, kernel_size=(3, 3),
                         activation='relu',
                         input_shape=data_train.image_shape))
        model.add(tf.contrib.keras.layers.Conv2D(64, (3, 3), activation='relu'))
        model.add(tf.contrib.keras.layers.MaxPooling2D(pool_size=(2, 2)))
        model.add(tf.contrib.keras.layers.Dropout(0.25))
        model.add(tf.contrib.keras.layers.Flatten())
        model.add(tf.contrib.keras.layers.Dense(128, activation='relu'))
        model.add(tf.contrib.keras.layers.Dropout(0.5))
        model.add(tf.contrib.keras.layers.Dense(4, activation='softmax'))
        
        model.compile(loss=tf.contrib.keras.losses.categorical_crossentropy,
                      optimizer=tf.contrib.keras.optimizers.Adadelta(),
                      metrics=['accuracy'])

        model.fit_generator(data_train,
                            steps_per_epoch=100,
                            epochs=5,
                            validation_data=data_test,
                            validation_steps= data_test.batch_size)

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by karmel** [comment]
@fchollet Any comments on the intended behavior here?

**Comment by tensorflowbutler** [comment]
Nagging Assignees @karmel, @fchollet: It has been 18 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @karmel, @fchollet: It has been 33 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @karmel, @fchollet: It has been 48 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @karmel, @fchollet: It has been 63 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @karmel, @fchollet: It has been 78 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by karmel** [comment]
@omalleyt12 -- is this something you can look into?

**Comment by omalleyt12** [comment]
@karmel yep I'll take a look

**Comment by omalleyt12** [comment]
@iustin94 is ```data_train``` a subclass of keras.Sequence? I'm guessing this has to do with multiprocessing

**Comment by iustin94** [comment]
I can't answer this question as I don't have access to that repository currently. You might be right, I recall we were attempting to use multiprocessing however we couldn't see it in the process monitor so we suspected we were doing something wrong.

**Comment by omalleyt12** [comment]
I tried to recreate the lack of error situation with the code below but this seems to be (correctly) throwing an error for me:

```python 
import numpy as np
import tensorflow as tf

keras = tf.keras

class MySequence(keras.utils.Sequence):
  def __getitem__(self, x):
    if x == 1:
      raise ValueError('Should raise an error.')
    return np.ones([10,100,100,3]), np.ones([10,4])

  def __len__(self):
    return 3

model = keras.models.Sequential()
model.add(keras.layers.Conv2D(32, kernel_size=(3, 3),
                              activation='relu',
                              input_shape=(100,100,3)))
model.add(keras.layers.Conv2D(64, (3, 3), activation='relu'))
model.add(keras.layers.MaxPooling2D(pool_size=(2, 2)))
model.add(keras.layers.Dropout(0.25))
model.add(keras.layers.Flatten())
model.add(keras.layers.Dense(128, activation='relu'))
model.add(keras.layers.Dropout(0.5))
model.add(keras.layers.Dense(4, activation='softmax'))

model.compile(loss=keras.losses.categorical_crossentropy,
              optimizer=keras.optimizers.Adadelta(),
              metrics=['accuracy'])

model.fit_generator(MySequence(),
                    steps_per_epoch=3,
                    epochs=1)
```

On the ```multiprocessing``` bit, I looked into it more and the code you had will use ```multithreading``` and not ```multiprocessing``` to grab the input data, which may explain why you weren't seeing anything in the process monitor. If you want Keras to use ```multiprocessing``` with a generator/Sequence, you can do:

```python
model.fit_generator(MySequence(),
                    steps_per_epoch=3,
                    epochs=1,
                    use_multiprocessing=True)
```

Let me know if there's any more context you can provide, otherwise I'm not sure if I'll be able to recreate this issue

**Comment by omalleyt12** [comment]
Closing for now, this issue may have been fixed since this bug was reported
